File: crates/codegraph-server/tests/fixtures/languages/sample.hs
module MyApp.User where

import Data.Text (Text)
import qualified Data.Map as Map

data User = User
  { userName :: Text
  , userEmail :: Text
  }

createUser :: Text -> Text -> User
createUser name email = User name email

greet :: User -> Text
greet user = "Hello, " <> userName user

File: crates/codegraph-server/tests/fixtures/languages/sample.pl
package MyApp::User;
use strict;
use warnings;
use Moose;

sub new {
    my ($class, %args) = @_;
    return bless \%args, $class;
}

sub greet {
    my ($self) = @_;
    print "Hello, " . $self->{name} . "\n";
}

sub _validate {
    my ($self) = @_;
    die "Name required" unless $self->{name};
}

1;
